torch/ao/quantization/experimental/APoT_tensor.py

```diff
@@ -1,14 +1,41 @@
 import torch
 from torch import Tensor
+from torch.ao.quantization.experimental.observer import APoTObserver, apot_to_float
 
 # class to store APoT quantized tensor
 class TensorAPoT(torch.Tensor):
     @staticmethod
     def quantize_APoT(tensor2quantize: Tensor) -> Tensor:
         raise NotImplementedError
 
-    def dequantize(self) -> Tensor:
-        raise NotImplementedError
+    """ Dequantizes integer Tensor to floating point representation
+    based on the calculated quantization levels from a specified APoT non-uniform observer.
+    The approach follows the method outlined in the APoT paper: https://arxiv.org/pdf/1909.13144.pdf.
+    Args:
+        tensor2dequantize: integer Tensor
+        b: total number of bits across all terms in non-uniform observer
+        k: base bitwidth, i.e. bitwidth of every term, in non-uniform observer
+    Returns:
+        result: floating point representation of input Tensor
+    """
+    @staticmethod
+    def dequantize(tensor2dequantize: Tensor, b: int, k: int) -> Tensor:  # type: ignore[override]
+        tensor2dequantize = tensor2dequantize.float()
+
+        # by paper defn, max value of floating point tensor will be 1.0
+        max_val = 1.0
+
+        # make observer
+        obs = APoTObserver(max_val=max_val, b=b, k=k)
```

Comment:
shouldn't dequantize take a quantized tensor as input? and should we store quantization parameters in the quantized tensor instead of using observer to get them?